File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.java
// Last updated: 5/29/2026, 11:56:56 AM
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] arr = new int[n];
        int leftp = 1;
        for (int i = 0; i < n; i++) {
            arr[i] = leftp;         
            leftp *= nums[i];
        }
        int rightp = 1;
        for (int i = n - 1; i >= 0; i--) {
            arr[i] *= rightp;       
            rightp *= nums[i];
        }
        return arr;
    }
}

File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.py
# Last updated: 5/29/2026, 11:56:50 AM
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n=len(nums)
        out=[1]*n
        lps=1
        rps=1
        for i in range(n):
            out[i]*=lps
            lps*=nums[i]
        for i in range(n-1,-1,-1):
            out[i]*=rps
            rps*=nums[i]
        return out